### PR TITLE
Audio test profile modifications 

### PR DESCRIPTION
**Pull request by ishthiyaqahmed**
Test profile modifications as per the yaml file.

### CODE REVIEW

**Comment by bhanucbp** [comment]
- test_l1_dsAudio_positive_dsGetAudioFormat, test_l1_dsAudio_negative_dsGetAudioFormat - dsGetAudioFormat is port independant. No need to loop through all the ports. Check the interface for other APIs

- no need of array of handles if you combine the loops. remove allocating array of handles.
use single handle and combine the loop
// allocate memory for the handles
handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
if(handle == NULL)
{
        UT_LOG_ERROR("Failed to allcoate memory for handlers");
        return;
}

**Review by bhanucbp — COMMENTED** [review]
Address review comments posted as comment

**Comment by ishthiyaqahmed** [comment]
> * test_l1_dsAudio_positive_dsGetAudioFormat, test_l1_dsAudio_negative_dsGetAudioFormat - dsGetAudioFormat is port independant. No need to loop through all the ports. Check the interface for other APIs
> * no need of array of handles if you combine the loops. remove allocating array of handles.
>   use single handle and combine the loop
>   // allocate memory for the handles
>   handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
>   if(handle == NULL)
>   {
>   UT_LOG_ERROR("Failed to allcoate memory for handlers");
>   return;
>   }

> * test_l1_dsAudio_positive_dsGetAudioFormat, test_l1_dsAudio_negative_dsGetAudioFormat - dsGetAudioFormat is port independant. No need to loop through all the ports. Check the interface for other APIs

done 

> * no need of array of handles if you combine the loops. remove allocating array of handles.
>   use single handle and combine the loop
will take care in code clean up activity

**Review by bhanucbp — DISMISSED** [review]
ship it

**Review by Sumip98 — DISMISSED** [review]
Looks good

File: src/test_l1_dsAudio.c [hunk 1]
```diff
@@ -4486,41 +5187,52 @@ void test_l1_dsAudio_negative_dsGetAudioLevel(void) {
  * |04|Call dsAudioPortTerm() - Terminate audio ports | | dsERR_NONE |  Termination must be successful |
  * 
  */
-void test_l1_dsAudio_positive_dsSetAudioLevel(void) {
+void test_l1_dsAudio_positive_dsSetAudioLevel_sink(void) {
```
Comment (hari22yuva) [inline src/test_l1_dsAudio.c:5190]: where is test case for source ?
dsSetAudioLevel()
check this 
https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAudio.h#L1057

File: src/test_l1_dsAudio.c [hunk 2]
```diff
@@ -4355,49 +5039,56 @@ void test_l1_dsAudio_negative_dsSetAudioGain(void) {
  * |06|Call dsAudioPortTerm() - Terminate audio ports | | dsERR_NONE |  Termination must be successful |
  * 
  */
-void test_l1_dsAudio_positive_dsGetAudioLevel(void) {
+void test_l1_dsAudio_positive_dsGetAudioLevel_sink(void) {
```
Comment (hari22yuva) [inline src/test_l1_dsAudio.c:5042]: where is the test case of source 
dsGetAudioLevel() 
https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAudio.h#L1033

**Review by hari22yuva — CHANGES_REQUESTED** [review]
i just checked only 2 API's it's not clearly not mention the test case for Source
i think we need to capture the test case of source as specified in the header file

File: src/test_l1_dsAudio.c [hunk 3]
```diff
@@ -3485,41 +4440,52 @@ void test_l1_dsAudio_positive_dsGetMS12AudioProfile(void) {
...
+        // Step 02: Get the port handle for all supported audio ports
+        for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+                result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
+                UT_ASSERT_EQUAL(result, dsERR_NONE);
+                UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
-			// Step 04: Verify that the profile string is valid and not empty
-			UT_ASSERT_NOT_EQUAL(strlen(currentProfile), 0);
-		}
+                if(handle[i] == dsAUDIOPORT_TYPE_SPEAKER) {
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:4465]: comparing handle with type is not correct

File: src/test_l1_dsAudio.c [hunk 4]
```diff
@@ -3550,45 +4516,56 @@ void test_l1_dsAudio_negative_dsGetMS12AudioProfile(void) {
...
+                UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
-	// Step 07: Attempt to get MS12 Audio Profile after termination
-	result = dsGetMS12AudioProfile(handle[0], currentProfile);
-	CHECK_FOR_EXTENDED_ERROR_CODE( result, dsERR_NOT_INITIALIZED, dsERR_INVALID_PARAM );
-	
-	// Logging at the end
-	UT_LOG("\n Out %s\n", __FUNCTION__);
+                // Step 05: Attempt to get MS12 Audio Profile with a null pointer
+                if(handle[i] == gDSAudioPortConfiguration[0].typeid ) {
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:4550]: comparing handle with typeid is not correct

File: src/test_l1_dsAudio.c [hunk 5]
```diff
@@ -826,34 +881,48 @@ void test_l1_dsAudio_positive_dsGetDialogEnhancement(void) {
...
-	result = dsAudioPortTerm();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // Step 03: Get the dialog enhancement levels for each port
+        for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+                if(gDSAudioPortConfiguration[i].ms12_capabilites & 0x04)
+                {
+                        result = dsGetDialogEnhancement(*(handle+i), &dialogEnhancementLevel);
+                        UT_ASSERT_EQUAL(result, dsERR_NONE);
+                        UT_ASSERT_TRUE(dialogEnhancementLevel >= 0 && dialogEnhancementLevel <= 16); // Valid level range check
+                }
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:914]: else part add check for NOT_SUPPORTED

File: src/test_l1_dsAudio.c [hunk 6]
```diff
@@ -949,40 +1029,58 @@ void test_l1_dsAudio_positive_dsSetDialogEnhancement(void) {
...
+                        // Step 05: Set Dialog Enhancement level - Mid DE Level(16)
+                        result = dsSetDialogEnhancement(*(handle+i), mid_de_level);
+                        UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-	// Step 06: Terminate audio ports
-	result = dsAudioPortTerm();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+                } else {
+                        result = dsSetDialogEnhancement(*(handle+i), max_de_level);
+                        UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
```
Comment (bhanucbp) [inline src/test_l1_dsAudio.c:1070]: check for NOT_SUPPORTED, not INVALID_PARAM
If capability not supported check for NOT_SUPPORTED. Check in other places as well

**Review by bhanucbp — COMMENTED** [review]
address the review comments

File: src/test_l1_dsAudio.c (hunk 6 above)
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c (hunk 5 above)
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:915]: done

File: src/test_l1_dsAudio.c (hunk 4 above)
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c (hunk 3 above)
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

**Review by bhanucbp — COMMENTED** [review]
ship it

File: src/test_l1_dsAudio.c [hunk 7]
```diff
@@ -5611,31 +7062,46 @@ void test_l1_dsAudio_positive_dsEnableLEConfig(void) {
...
+        // Steps 02 to 03: Enable LE Config for each port
+        for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+                result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
+                UT_ASSERT_EQUAL(result, dsERR_NONE);
+                UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
-	}
+                // Enable LE Config
+                result = dsEnableLEConfig(*(handle+i), true);
+             if((gSourceType == 0) && (gDSAudioPortConfiguration[i].ms12_capabilites & 0x400)) {
```
Comment (hari22yuva) [inline src/test_l1_dsAudio.c:7088]: While clean the code please take care replace 0x400 to corresponding enum values

**Review by bhanucbp — APPROVED** [review]

**Pull request merged**